API/SearchService/src/util.py

```diff
@@ -0,0 +1,75 @@
+from enum import Enum
+import azure.functions as func
+
+from .exception import ParameterException
+
+class SearchType(int, Enum):
+    All = 1
+    Scenario = 2
+    Command = 3
+
+    def get_search_fields(self):
+        if self == self.All:
+            return ["id", "description", "commandSet/command"]
+        elif self == self.Scenario:
+            return ["id", "description"]
+        elif self == self.Command:
+            return ["commandSet/command"]
+
+
+def get_param(req: func.HttpRequest, name: str, required=True, default=None):
+    value = req.params.get(name)
+    if not value:
+        try:
+            req_body = req.get_json()
+            value = req_body.get(name)
+        except ValueError:
+            pass
+    if required and value is None:
+        raise ParameterException(f'Illegal parameter: please pass in the parameter "{name}"')
+    elif value is None:
+        return default
+    return value
```

Comment:
It is recommended to unbind the `required` and `default` parameters.
When `if required and value is None` but `default is not None`, instead of raising ParameterException, it should assign `default` to `value`

Reply:
In fact, different method can have different default `default`. 
For example, the `default` of `get_param_str` defaults to `''`. If `required` set to `True` and `default` not set, default `default`, `''` will be passed in. In such case, we still should raise a `ParameterException`.